client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+import { setFocusHideClass } from 'lib/layout-focus';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( ! isValidArea( action.area ) ) {
+				return state;
+			}
+			setFocusHideClass();
```

Comment:
I took a closer look, and I don't think this class is even used anymore ([search](https://github.com/Automattic/wp-calypso/search?utf8=%E2%9C%93&q=focus-hide)). In fact, the original issue that this was trying to solve is now present once again, i.e. using browser search on sites section matches sites in Site picker even when site picker is not open (originally 2753-gh-calypso-pre-oss, cc @mtias). 

If we _were_ to reintroduce this, I wonder if we could do it by monitoring the `currentLayout` state from the sidebar component itself to add and remove the class.

In the meantime, I think we should just remove all traces of `setFocusHideClass`.